### PR TITLE
Test to make sure JPY PyObject.getXXXValue() errors out immediately

### LINKED ISSUE #3597: Add  a JPY test case to verify a fix that the next JPY release will include
The fix targets the issue reported in https://github.com/jpy-consortium/jpy/issues/98

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #3597

Linked to: https://github.com/jpy-consortium/jpy/pull/99

### CODE REVIEW

**Review by chipkent — DISMISSED** [review]

File: py/jpy-integration/src/javaToPython/java/io/deephaven/jpy/integration/ErrorOutTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+public class ErrorOutTest extends PythonTest {
+    // ----- error out -----
+    @Test(expected = RuntimeException.class)
+    @Ignore("Need to wait for the next JPY release to fix https://github.com/jpy-consortium/jpy/issues/98")
+    public void errorIntOutFromNone() {
+        String expr = "[None, []]";
+        PyObject in = PyObject.executeCode(expr, PyInputMode.EXPRESSION);
+        in.asList().get(0).getIntValue();
```
Comment (devinrsmith) [inline py/jpy-integration/src/javaToPython/java/io/deephaven/jpy/integration/ErrorOutTest.java:23]: I'd like to make sure we have tests that demonstrate getIntValue w/ python ints in the cases:

* 0-32 bits
* 33-64 bits
* >64 bits

Given the new code, I expect the first two cases to "work", and the last case to properly fail.

File: py/jpy-integration/src/javaToPython/java/io/deephaven/jpy/integration/ErrorOutTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Test(expected = RuntimeException.class)
+    @Ignore("Need to wait for the next JPY release to fix https://github.com/jpy-consortium/jpy/issues/98")
+    public void errorLongOutFromNone() {
+        String expr = "[None, []]";
+        PyObject in = PyObject.executeCode(expr, PyInputMode.EXPRESSION);
+        in.asList().get(0).getLongValue();
```
Comment (devinrsmith) [inline py/jpy-integration/src/javaToPython/java/io/deephaven/jpy/integration/ErrorOutTest.java:31]: I'd like to make sure we have tests that demonstrate getLongValue w/ python ints in the cases:

* 0-64 bits
* >64 bits

Given the new code, I expect the first case to work and the last case to properly fail.

**Comment by devinrsmith** [comment]
Given the lossy nature of getIntValue(), we should audit all of our own uses of it and switch over to more precise calls:

```java
Math.toIntExact(valueIn.getLongValue())
```

or something equivalent to that (considering int, short, byte, and char consumer).

**Comment by devinrsmith** [comment]
Follow-up https://github.com/deephaven/deephaven-core/issues/3716

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**